assets/js/googlesitekit/datastore/forms/index.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * core/forms data store
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import invariant from 'invariant';
+import { STORE_NAME } from './constants';
+
+export { STORE_NAME };
+
+const SET_FORM_VALUES = 'SET_FORM_VALUES';
+
+export const INITIAL_STATE = {};
+
+export const actions = {
+	/**
+	 * Stores site form information.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {string} formName Name of the form.
+	 * @param {Object} formData Form data supplied as a map of keys to set and their respective values.
+	 * @return {Object} Redux-style action.
+	 */
+	setValues( formName, formData ) {
+		invariant( formName, 'formName is required for setting values.' );
+		invariant( formData instanceof Object && formData.constructor === Object, 'formData must be an object.' );
```

Comment:
I think checking truthiness is enough here; the docs spell out the usage but this sort of stuff seems fragile and not worth running on every call to me. 🤷‍♂️

Reply:
As agreed this check is to be left as is